Sources/Datadog/RUMMonitor.swift

```diff
@@ -607,6 +608,25 @@ public class RUMMonitor: DDRUMMonitor, RUMCommandSubscriber {
                 }
             }
         }
+
+        // update the core context with rum context
+        core.set(feature: "rum", attributes: self.queue.sync {
+            let applicationContext = self.applicationScope.context
+            let sessionContext = self.applicationScope.sessionScope?.context
+            let activeViewContext = self.applicationScope.sessionScope?.viewScopes.last?.context
+
+            guard let sessionContext = sessionContext, sessionContext.sessionID != .nullUUID else {
+                // if Session was sampled or not yet started
+                return [:]
+            }
```

Comment:
Would it be possible to reuse [`RUMCurrentContext`](https://github.com/DataDog/dd-sdk-ios/blob/21fa77710f6fb6b0d8dfd17b24eec817163c57ae/Sources/Datadog/RUM/RUMContext/RUMCurrentContext.swift#L10) somehow in there? It does similar computation, with slight differences (it leverages `??`) and is already fully tested.

Reply:
I was planning on removing `RUMCurrentContext`, so the `queue.sync` always appears in the queue's owner (= RUMMonitor) and not down the stream. But I could move this logic to a dedicated private method, WDYT?

Reply:
Also, `RUMContextProvider` is still there because of WebView integration. But I plan on removing it once it's connected to the message bus.

Reply:
> I was planning on removing RUMCurrentContext, so the queue.sync always appears in the queue's owner

Got it, sounds good as long as we're aware of it and will revamp it later 👍. And we won't miss this change as compiler will spot it when extracting WVs integration later. All good 👌.